src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -555,13 +560,37 @@ private void transferAlignmentInfoToFragment(final SAMRecord unaligned, final SA
             log.warn("Record mapped off end of reference; making unmapped: " + aligned);
             SAMUtils.makeReadUnmapped(unaligned);
         } else if (isContaminant) {
-            log.warn("Record looks like foreign contamination; making unmapped: " + aligned);
-            // NB: for reads that look like contamination, just set unmapped flag and zero MQ but keep other flags as-is.
-            // this maintains the sort order so that downstream analyses can use them for calculating evidence
-            // of contamination vs other causes (e.g. structural variants)
+            if (numForeignContaminantWarnings < MAX_FOREIGN_CONTAM_WARN) {
+                log.warn("Record looks like foreign contamination ; making unmapped: " + aligned);
+                numForeignContaminantWarnings++;
+            }
+
+            if (numForeignContaminantWarnings == MAX_FOREIGN_CONTAM_WARN) {
+                log.warn("No more warnings regarding foreign contamination; limit reached.");
+            }
+
+            // For reads that look like contamination, set unmap. And reset the mapping information as
+            // keeping it violates the sam-spec and is not supported by some formats (CRAM)
+
+            // Save the old alignment information in a "PA" Tag in case it is needed.
+            unaligned.setAttribute("PA", String.join(",",
+                    aligned.getContig(),
+                    ((Integer) aligned.getAlignmentStart()).toString(),
+                    aligned.getCigarString(),
+                    ((Integer) aligned.getMappingQuality()).toString(),
+                    aligned.getStringAttribute(SAMTag.NM.name()))+";");
+
             unaligned.setReadUnmappedFlag(true);
             unaligned.setMappingQuality(SAMRecord.NO_MAPPING_QUALITY);
-            unaligned.setAttribute(SAMTag.FT.name(), "Cross-species contamination");
+            unaligned.setReferenceIndex(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX);
+            unaligned.setAlignmentStart(0);
+            unaligned.setCigar(null);
+            unaligned.setCigarString(SAMRecord.NO_ALIGNMENT_CIGAR);
```

Comment:
Do SV callers use the fact that the record is unmapped but has coordinates?  The previous comment makes me concerned that we could be breaking callers relying on this behavior. 👎